zio-http/shared/src/main/scala/zio/http/URL.scala

```diff
@@ -344,21 +345,27 @@ object URL {
   }
 
   private def encode(url: URL): String = {
-    def path(relative: Boolean) =
-      QueryParamEncoding.default.encode(
-        if (relative || url.path.isEmpty) url.path.encode else url.path.addLeadingSlash.encode,
-        url.queryParams.normalize,
+    val path =
+      QueryParamEncoding.encode(
+        url.path.encodeBuilder,
+        url.queryParams,
         Charsets.Http,
-      ) + url.fragment.fold("")(f => "#" + f.raw)
+      )
+
+    def fromAbsURL(abs: Location.Absolute, path: String) = {
```

Comment:
Scala's string interpolator is terrible performance-wise because it initializes the string builder with an initial size equal to the number of constant chars.

If this is in the hotpath I recommend using a StringBuilder and initializing it with a pre-calculated initial value